### PR TITLE
Separate edit YAML page

### PR DESCRIPTION
**Pull request by spadgett**
This PR changes the YAML editor to be a full page rather than a dialog. This gives the editor more space as well as making it bookmarkable.

Fixes https://github.com/openshift/origin/issues/6845

screen shot 2016-09-22 at 11 28 21 am

@jwforres PTAL

### CODE REVIEW

File: app/scripts/controllers/edit/yaml.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      title: "Edit YAML"
+    }];
+
+    var navigateBack = function() {
+      if ($routeParams.returnURL) {
+        $location.url($routeParams.returnURL);
+        return;
+      }
+
+      // Default to going back in histroy if no returnURL.
```
Comment (jwforres) [inline app/scripts/controllers/edit/yaml.js]: typo, history

File: app/scripts/controllers/edit/yaml.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+          group: $routeParams.group
+        };
+
+        DataService.get(resourceGroupVersion, $scope.name, context).then(
+          function(result) {
+            // Modify a copy of the resource.
+            var resource = angular.copy(result);
+            $scope.resource = resource;
+
+            // TODO: Update the BreadcrumbsService to handle types without browse pages.
```
Comment (jwforres) [inline app/scripts/controllers/edit/yaml.js]: remove the TODO comment

File: app/scripts/controllers/edit/yaml.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            // order, however, so if we add apiVersion and kind first, they appear at
+            // the top of the serialized YAML. The rest of the properties are in the
+            // order returned from the API that we want (metadata, spec, status).
+            resource = angular.extend({
+              apiVersion: resource.apiVersion,
+              kind: resource.kind
+            }, resource);
+
+            // flowLevel: 8 - Maximum level of depth before generating inline
+            _.set($scope, 'editor.model', jsyaml.safeDump(resource, {flowLevel: 8}));
```
Comment (jwforres) [inline app/scripts/controllers/edit/yaml.js]: bump to 10

File: app/scripts/controllers/edit/yaml.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                                              ProjectsService) {
+    if (!$routeParams.kind || !$routeParams.name) {
+      Navigate.toErrorPage("Kind or name parameter missing.");
+      return;
+    }
+
+    var humanizeKind = $filter('humanizeKind');
+    $scope.name = $routeParams.name;
+    $scope.resourceURL = Navigate.resourceURL($scope.name, $routeParams.kind, $routeParams.project);
+    $scope.breadcrumbs = [{
```
Comment (jwforres) [inline app/scripts/controllers/edit/yaml.js]: fix navigateBack in the breadcrumb for other resources page

File: app/scripts/controllers/edit/yaml.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                  };
+                });
+            };
+
+            $scope.cancel = function() {
+              navigateBack();
+            };
+
+            // Watch for changes to warn the user.
+            watches.push(DataService.watchObject(resourceGroupVersion, $scope.name, context, function(newValue, action) {
```
Comment (jwforres) [inline app/scripts/controllers/edit/yaml.js]: handle unsupported watch errors, e.g. on RoleBindings

File: app/scripts/controllers/otherResources.js [hunk 6]
```diff
@@ -41,6 +41,19 @@ angular.module('openshiftConsole')
       }
     });
 
+    $scope.getReturnURL = function() {
+      var kind = _.get($scope, 'kindSelector.selected.kind');
+      if (!kind) {
+        return '';
+      }
+
+      return URI.expand("project/{projectName}/browse/other?kind={kind}&group={group}", {
```
Comment (jwforres) [inline app/scripts/controllers/otherResources.js]: This makes me wonder if we should be changing the URL state when you pick something in the dropdown, to make those lists bookmarkable

File: app/scripts/controllers/otherResources.js [hunk 7]
```diff
@@ -65,6 +78,12 @@ angular.module('openshiftConsole')
         $scope.project = project;
         $scope.context = context;
         $scope.kindSelector.disabled = false;
+
+        // Optional query param to preselect a kind.
+        if ($routeParams.kind) {
```
Comment (jwforres) [inline app/scripts/controllers/otherResources.js]: what happens if someone gives you a bogus kind/group param

File: app/scripts/filters/util.js [hunk 8]
```diff
@@ -418,6 +418,11 @@ angular.module('openshiftConsole')
       return url;
     };
   })
+  .filter('editYamlURL', function(Navigate) {
```
Comment (jwforres) [inline app/scripts/filters/util.js]: we are getting enough of these i'm starting to think they need their own filters/navigate.js, not necessarily in this PR though

File: app/scripts/controllers/edit/yaml.js (hunk 2 above)
Comment (spadgett) [inline app/scripts/controllers/edit/yaml.js]: But the TODO still applies

**Comment by spadgett** [comment]
@jwforres updated

**Comment by jwforres** [comment]
[merge]

**Pull request merged**